src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -633,6 +635,19 @@ int XrdHttpProtocol::Process(XrdLink *lp) // We ignore the argument here
           sbio = CreateBIO(Link);
           BIO_set_nbio(sbio, 1);
           ssl = SSL_new(sslctx);
+
+          // On newer versions of OpenSSL, we use the periodically
+          // updated store `verify_store` with the `x509_store_lock`
+          // held; older OpenSSL versions are missing `SSL_set1_*_cert_store`,
+          // so we simply rely on the global one in SSL_CTX.  The latter
+          // doesn't refresh the verify store, which causes issues with some
+          // plugins.
+#if OPENSSL_VERSION_NUMBER >= 0x010100000L
```

Comment:
Not so fast, the test should be '>' not '>=' because OpenSSL 1.01 does not have the set1 functions. I know from your comment that you did compile it on SL6 but that only worked because someone must have upgraded their OpenSSL (which of course they should have). Unfortunately, lots of sites use RH6 out of the box and it comes with 1.0.1e-fips and it definitely won't compile! I think all of teh test should change and hen you'll be fine.

Reply:
I'm not sure I understand this?

`OPENSSL_VERSION_NUMBER >= 0x010100000L` should test for 1.1.0 or later.  That is, it only enables this code if you are at least on 1.1.0 -- that should exclude any variant of RHEL6.

(FWIW - the OpenSSL RPM I tested against was `openssl-1.0.1e-58.el6_10.x86_64`)

Reply:
I'm not sure if it is intentional, but it will also exclude any variant of RHEL7, as e.g. both Centos7 and CC7 have openssl `1.0.2k`. From what I see `SSL_set1_verify_cert_store` & `SSL_set1_chain_cert_store` are available in OpenSSL 1.0.2 (see e.g. https://github.com/openssl/openssl/blob/OpenSSL_1_0_2k/ssl/ssl.h). If we would like to include RHEL7 we should test for `OPENSSL_VERSION_NUMBER >= 0x10002000L`.

Reply:
Argh - sorry, I haven't been able to sit down and get to this in the last two days (which have each had >6hrs of Zoom meetings...).  You're right, this is too strict.